### PR TITLE
feat(kuma-cp): allow Mesh default resources regeneration without deletion and restart

### PR DESCRIPTION
**Pull request by michaelbeaumont**
Feels more like this is how the annotation should work, given that we need it in the first place. There's no reason to filter only for `Create` events, which requires a restart to retrigger reconciliation.

WDYT?

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go [hunk 1]
```diff
@@ -53,31 +47,15 @@ func (r *MeshDefaultsReconciler) Reconcile(ctx context.Context, req kube_ctrl.Re
 		mesh.GetMeta().(*k8s.KubernetesMetaAdapter).Annotations = map[string]string{}
 	}
 	mesh.GetMeta().(*k8s.KubernetesMetaAdapter).GetAnnotations()[common_k8s.K8sMeshDefaultsGenerated] = "true"
-	r.Log.Info("marking mesh that default resources were generated", "mesh", req.Name)
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go:56]: I find this log useful

Comment (michaelbeaumont) [reply pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go:56]: My thinking was you either get an error or you can see on the mesh object that it has the annotation so the message is sort of superfluous in terms of ensuring that the controller is functioning

Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go:56]: The difference is that with a log you have this event on the timeline. If you investigate an issue using logs/metrics, it's good to see changes made by the system.

Comment (michaelbeaumont) [reply pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go:56]: Hmm that seems like something that could apply to anything the CP does, why is this creation of the default resources so important that it deserves so many log lines on Info? 

```
2023-03-07T12:37:52.445Z	INFO	controllers.mesh-defaults	ensuring that default mesh resources exist	{"mesh": "default"}
2023-03-07T12:37:52.445Z	INFO	defaults.mesh	ensuring default resources for Mesh exist	{"mesh": "default"}
2023-03-07T12:37:52.546Z	INFO	defaults.mesh	default Retry already exists	{"mesh": "default", "name": "retry-all-default"}
2023-03-07T12:37:52.646Z	INFO	defaults.mesh	default TrafficPermission already exists	{"mesh": "default", "name": "allow-all-default"}
2023-03-07T12:37:52.747Z	INFO	defaults.mesh	default TrafficRoute already exists	{"mesh": "default", "name": "route-all-default"}
2023-03-07T12:37:52.848Z	INFO	defaults.mesh	default Timeout already exists	{"mesh": "default", "name": "timeout-all-default"}
2023-03-07T12:37:52.949Z	INFO	defaults.mesh	default CircuitBreaker already exists	{"mesh": "default", "name": "circuit-breaker-all-default"}
2023-03-07T12:37:52.949Z	INFO	defaults.mesh	Dataplane Token Signing Key already exists	{"mesh": "default"}
2023-03-07T12:37:52.949Z	INFO	controllers.mesh-defaults	marking mesh that default resources were generated	{"mesh": "default"}
```

All resources get a line even though they have a creation timestamp. We even have an earlier message "ensuring that default..." right before this. It seems like one line plus the presence or absence of the annotation/resources would then be more than enough to debug an issue. But this PR isn't about log messages ultimately so I'll leave them alone

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go:56]: > Hmm that seems like something that could apply to anything the CP does

If that's an update operation on the object then yes.

We modify Mesh object to mark this event, I feel like this is somewhat useful in the end.

**Pull request merged**